### PR TITLE
I have yet to meet man who can outsmart bullet (The SG Minigun)

### PR DESCRIPTION
**Pull request by Arctrooper8**
## About The Pull Request

image
This is a reopening of MJP's PR #10024
However it is slightly different in certain ways.
The most noticeable difference lies in the SG nomenclature, which now uses an SG (clever I know) instead of the strange T.
The second biggest difference is the fact that the SG-85 has a much much faster firerate (even faster than minigun, it's hilarious) and weaker bullets to compensate this truly suppressive firerate. (I don't know how to record videos please help)
**Of course values are subject to change and discussion, helpful feedback is appreciated.**

### Video
https://imgur.com/a/mwKVjxY

## Why It's Good For The Game

First and foremost: Fun, it's a lot of fun to wield a gun that fires this fast, even if the overall DPS is quite similar to the SG-29. 

Second and more importantly: the SG is supposed to be a backline support gunner that provides suppressive fire against Xenos. The SG-25 does not fit this role well, being an IFF assault rifle with a low magazine size (compared to normal smartguns) and faster usage and movement speed which promotes chasing Xenos. 
The SG-85 fits the role much better due to it's tyrannical firerate and high ammo count, if the SG-85 user attempts to run off on their own they're likely to be stunned and have to spend time reconnecting their minigun and then winding it up again.

The SG-25 supports a lone wolf chasedown style that contrasts with the SG's role.
Whereas the SG-85 encourages cooperation and teamwork whilst providing ample amounts of supporting fire.

TLDR:
### BRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRT
(It's a ton of fun, trust me)

## Changelog
:cl: Arctrooper8 Code: MJP Sprites: Kuro
add: SG-85 handheld gatling gun, an SG minigun with an absurd firerate at the cost of low damage per bullet.
del: T-25 has truly been sent to the shadow realm this time (They will not survive the Brazilians).
balance: SG-29 now has a firerate of .2 instead of .25
/:cl:

### CODE REVIEW

**Comment by Walarks** [comment]
But can it hurt a georger... ?

**Comment by Arctrooper8** [comment]
> But can it hurt a georger... ?

https://www.youtube.com/watch?v=BGJhpeqn_6I&ab_channel=TeamFortress2-AllSounds

**Comment by breadhunt** [comment]
Portable crash machine go brr
I fear this man

**Comment by Arctrooper8** [comment]
> Portable crash machine go brr I fear this man

What?

**Comment by ilikesaying** [comment]
> > Portable crash machine go brr I fear this man
> 
> What?

The sheer quantity o f landed shots and maybe messages may crash low end clients

**Comment by Arctrooper8** [comment]
> > > Portable crash machine go brr I fear this man
> > 
> > 
> > What?
> 
> The sheer quantity o f landed shots and maybe messages may crash low end clients

Oops

**Review by Kuro020 — CHANGES_REQUESTED** [review]
image
? also the minigun sprite/mob sprites/power pack overall need more work

**Comment by Arctrooper8** [comment]
> image ? also the minigun sprite/mob sprites/power pack overall need more work

In the previous PR Tivi said to just entirely remove the T-25 instead of leaving remnants.
And I just used the fancy powerpack version sprite, what would you suggest to make it better?

File: code/modules/projectiles/gun_attachables.dm [hunk 1]
```diff
@@ -499,7 +499,7 @@ inaccurate. Don't worry if force is ever negative, it won't runtime.
 	flags_attach_features = NONE
 
 /obj/item/attachable/t29barrel
```
Comment (magatsuchi) [inline code/modules/projectiles/gun_attachables.dm:501]: paths should be updated to match their correct designation now (sg29barrel instead of t29 and so on)

File: html/changelogs/archive/2021-08.yml [hunk 2]
```diff
@@ -569,7 +569,7 @@
   Lewdcifer:
   - balance: Reverts Powerfist Stun.
   Losenis:
-  - balance: T-29 Smartmachinegun can now mount miniscope
+  - balance: SG-29 Smartmachinegun can now mount miniscope
```
Comment (magatsuchi) [inline html/changelogs/archive/2021-08.yml:572]: lmao please don't change any *.ymls
these are for referencing, doesn't impact the actual game

**Review by magatsuchi — CHANGES_REQUESTED** [review]

File: html/changelogs/archive/2021-08.yml (hunk 2 above)
Comment (Arctrooper8) [reply html/changelogs/archive/2021-08.yml:572]: Oops, that's what I get for using replace all

File: code/modules/projectiles/gun_attachables.dm (hunk 1 above)
Comment (Arctrooper8) [reply code/modules/projectiles/gun_attachables.dm:501]: I'll do that if I have extra time

File: html/changelogs/archive/2021-08.yml (hunk 2 above)
Comment (Arctrooper8) [reply html/changelogs/archive/2021-08.yml:572]: It might be for the best if people want to look back though, since in the future it wouldn't be called the T-29 and if they wanted to check the weapon history they'd have difficulty finding it.

**Review by Pariah919 — APPROVED** [review]
Brrrrrrtt.
Looks okay to me, might need new balance if it crashes games though.

File: code/modules/projectiles/gun_attachables.dm [hunk 3]
```diff
@@ -498,9 +498,9 @@ inaccurate. Don't worry if force is ever negative, it won't runtime.
 	icon_state = "t12barrel"
 	flags_attach_features = NONE
 
-/obj/item/attachable/t29barrel
-	name = "T-29 barrel"
-	icon_state = "t29barrel"
+/obj/item/attachable/sg29barrel
```
Comment (TiviPlus) [inline code/modules/projectiles/gun_attachables.dm:501]: no numbers in typepaths please

File: code/modules/projectiles/gun_attachables.dm [hunk 4]
```diff
@@ -972,10 +972,10 @@ inaccurate. Don't worry if force is ever negative, it won't runtime.
 	scatter_mod = 0
 	movement_acc_penalty_mod = 0
 
-/obj/item/attachable/stock/t29stock
-	name = "T-29 stock"
+/obj/item/attachable/stock/sg29stock
```
Comment (TiviPlus) [inline code/modules/projectiles/gun_attachables.dm:975]: ditto

**Review by TiviPlus — CHANGES_REQUESTED** [review]
rest lgtm

**Comment by Arctrooper8** [comment]
> rest lgtm

Fixed, ready to merge

**Review by TiviPlus — APPROVED** [review]

**Comment by Tailiik** [comment]
Why not make the SG-25 a req weapon instead of just entirely deleting a fun gun? Returning SG to needing a clunky backpack like the smartguns from hundreds of years earlier in the setting is lame because people are coping that one smartgun has some mobility to it
